### PR TITLE
feat: harden public API surface with sealed types, immutability, and approval tests

### PR DESCRIPTION
**Pull request by chrisjainsley**
## What

Hardens the public API surface across all Augustus packages by sealing classes, making types immutable, adding API approval tests, and cleaning up misleading/duplicate methods.

## Why

Without these protections, users could inherit from types not designed for extension, mutate supposedly-stable state (e.g. RouteInstruction.Pattern after construction breaks regex matching), and accidental breaking changes could ship undetected. This establishes a stable API contract enforced by tests.

## Services & Areas Changed
- Augustus (core) — sealed types, immutable RouteInstruction/RouteConfiguration, DynamicContentFields encapsulation, CacheOnly setter, Validate() cleanup, deprecated WithJsonResponse/WithDelay
- Augustus.AI — sealed types, removed fragile IReadOnlyCollection casts, widened CacheKeyComputer params
- Augustus.APIs.Stripe — sealed all 21 public classes
- Augustus.Reqnroll — sealed AugustusScenarioHooks
- All test projects — added PublicApiGenerator + Verify API approval tests

## Testing

- PublicApiGenerator + Verify snapshot tests added for Augustus, Augustus.AI, and Augustus.APIs.Stripe
- All 379 existing tests pass across net9.0 and net10.0
- CacheOnlyModeTests updated to validate setter-based enforcement

## Breaking Changes

- All public classes are now `sealed` — any subclasses will fail to compile
- `RouteInstruction.Pattern`, `HttpMethod`, `Instructions` are now read-only
- `RouteConfiguration.ResponseStrategy` changed from `set` to `init`
- `APISimulatorOptions.DynamicContentFields` changed from `List<string>` to `IList<string>`
- Implicit operator removed from `InstructionBuilder` — callers must use `.Build()`
- `APISimulatorOptions.Validate()` no longer mutates state (CacheOnly setter enforces invariants)
- Augustus.AI and Augustus.Stripe versions aligned to 0.0.1

## API Changes

```csharp
// RouteInstruction — now immutable after construction
var ri = new RouteInstruction("/api/test/{id}", "GET");
// ri.Pattern = "x";  // Compile error — read-only
// ri.Instructions.Add("x");  // Compile error — IReadOnlyList

// RouteBuilder — WithJsonResponse deprecated
simulator.ForGet("/api/x")
    .WithResponse("{}")  // Use this instead of WithJsonResponse
    .Add();

// InstructionBuilder — explicit Build() required
simulator.ConfigureRoutes()
    .ForGet("/api/x")
    .WithInstruction("Return 200")
    .Build();  // Was previously implicit via operator

// DynamicContentFields — IList<string> instead of List<string>
options.DynamicContentFields.Add("timestamp");  // Still works
// options.DynamicContentFields = new List<string>();  // Compile error
```

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: Augustus/Augustus/APISimulatorOptions.cs [hunk 1]
```diff
@@ -111,28 +125,25 @@ public int Port
...
     public void Validate()
     {
-        if (CacheOnly)
-        {
-            EnableCaching = true;
-            AutoRemoveStaleCache = false;
-        }
+        // Currently no validation rules beyond what setters enforce.
+        // Cache-only invariants are enforced by the CacheOnly property setter.
     }
```
[the remark this answers is not in the document]
Comment (chrisjainsley) [reply Augustus/Augustus/APISimulatorOptions.cs]: Fixed in 8f12360. \ now enforces \ cross-property invariants (\ must be true, \ must be false), throwing \ on violations. The setter still applies defaults as a convenience, and \ catches post-setter overrides.

File: Augustus/Augustus/CacheKeyBodyNormalizer.cs [hunk 2]
```diff
@@ -7,9 +7,9 @@ internal static class CacheKeyBodyNormalizer
 {
     private const string NormalizedPlaceholder = "__NORMALIZED__";
 
-    public static byte[] NormalizeForCacheKey(byte[] body, IReadOnlyCollection<string> propertyNames)
+    public static byte[] NormalizeForCacheKey(byte[] body, IEnumerable<string> propertyNames)
     {
-        if (propertyNames.Count == 0 || body.Length == 0)
+        if (!propertyNames.Any() || body.Length == 0)
             return body;
```
[the remark this answers is not in the document]
Comment (chrisjainsley) [reply Augustus/Augustus/CacheKeyBodyNormalizer.cs]: Fixed in 8f12360. Changed the parameter type from \ to \. The caller (\) materializes once, and \ receives a materialized collection throughout recursion — no more repeated enumeration.

File: Augustus/Augustus/RouteInstruction.cs [hunk 3]
```diff
@@ -46,11 +48,16 @@ public class RouteInstruction
     /// <param name="httpMethod">The HTTP method to match, or "*" for all methods. Default is "*".</param>
     public RouteInstruction(string pattern, string httpMethod = "*")
     {
-        Pattern = pattern;
+        Pattern = pattern ?? throw new ArgumentNullException(nameof(pattern));
         HttpMethod = httpMethod.ToUpperInvariant();
         CompilePattern();
```
[the remark this answers is not in the document]
Comment (chrisjainsley) [reply Augustus/Augustus/RouteInstruction.cs]: Fixed in 8f12360. Added \ before the \ call. Now throws a clear \ instead of a \.
